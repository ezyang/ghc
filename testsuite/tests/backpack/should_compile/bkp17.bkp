unit p where
    signature H where
unit q where
    module M where
unit r where
    include q (M as H)
    include p
