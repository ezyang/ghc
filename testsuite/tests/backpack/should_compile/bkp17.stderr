[1 of 3] Processing p
/--- Shape for p
provides:
requires: H -> p(hole:H):H
signature p(hole:H):H
\---
  [1 of 1] Compiling H[abstract sig]  ( p/H.hsig, nothing )
[2 of 3] Processing q
  Instantiating q()
/--- Shape for q
provides: M -> q():M
requires:
module q():M
\---
  [1 of 1] Compiling M                ( q/M.hs, bkp17-out/Br2hqQttvyiBkpnTZJaMkb/M.o )
[3 of 3] Processing r
  Instantiating r()
/--- Shape for r
provides: H -> q():M
requires:
include q() [(H, q():M)] requires []
include p(H -> q():M) [] requires [(H, [p(H -> q():M):H])]
\---
  [1 of 2] Including q
    Instantiating q()
  [2 of 2] Including p
    Instantiating p(H -> q():M)
/--- Shape for p
provides:
requires: H -> p(H -> q():M):H
signature p(H -> q():M):H
\---
    [1 of 1] Compiling H[abstract sig]  ( p/H.hsig, nothing )
